### PR TITLE
Gestures on PDFs

### PR DESCRIPTION
**Pull request by nucleogenesis**
### Summary

Adds [hammerjs](http://hammerjs.github.io/) and in PDF Renderer make use of it to handle pinch-zoom gestures on mobile.

Does not work on iOS because Apple's pinch zooming always zooms the whole window without regard for any kind of meta viewport settings (ie, `maximum-scale` or `user-scalable`). 

However, it works on Android.

Basically, a pinch will call the same `zoomIn` or `zoomOut` function as tapping the + and - buttons do. They're throttled to 1 second to avoid accidentally zooming all the way in with one pinch.

NOTE: Not tested in a real life proper "app context" running the Android app - but still a valuable addition for non-app users who do use android devices anyway.

### Reviewer guidance

Open a PDF using an Android device and pinch zoom in and out.

Testing this without an actual device is a pain - but if you use Genymotion [read this](https://docs.genymotion.com/desktop/3.0/03_Virtual_devices/035_Generating_virtual_device_logs.html) (it wasn't perfect for me, but good enough early on) 

### References

https://github.com/learningequality/kolibri/issues/3710

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by indirectlylit** [comment]
@mrpau-dev would you mind testing this on Android?

File: kolibri/plugins/pdf_viewer/assets/src/views/PdfRendererIndex.vue [hunk 1]
```diff
@@ -122,6 +123,28 @@
...
+
+        if (navigator.platform) {
+          while (iDevices.length) {
+            if (navigator.platform === iDevices.pop()) {
+              return true;
+            }
+          }
+        }
+
+        return false;
```
Comment (indirectlylit) [inline kolibri/plugins/pdf_viewer/assets/src/views/PdfRendererIndex.vue]: can simplify a bit:

```suggestion
        const iDevices = [
          'iPad Simulator',
          'iPhone Simulator',
          'iPod Simulator',
          'iPad',
          'iPhone',
          'iPod',
        ];
        return iDevices.includes(navigator.platform);
```

File: kolibri/plugins/pdf_viewer/assets/src/views/PdfRendererIndex.vue [hunk 2]
```diff
@@ -122,6 +123,28 @@
     }),
     computed: {
       ...mapGetters(['sessionTimeSpent']),
+      // Returns whether or not the current device is iOS.
+      // Probably not perfect, but worked in testing.
+      iOS() {
+        var iDevices = [
```
Comment (indirectlylit) [inline kolibri/plugins/pdf_viewer/assets/src/views/PdfRendererIndex.vue]: FYI even if you wanted to `pop` this, can still use `const` instead of `var`

**Comment by mrpau-richard** [comment]
@indirectlylit I'm having the same [issue](https://github.com/learningequality/kolibri-installer-android/issues/65) with the installer built in this PR. The Kolibri Android app crashes after I launch it.

**Comment by indirectlylit** [comment]
merging to get it in place for integration testing

**Pull request merged**